### PR TITLE
Adding notifications of data changed

### PR DESCRIPTION
**Pull request by giladgo**

### CODE REVIEW

File: cloud/main.js [hunk 1]
```diff
@@ -237,7 +237,11 @@ Parse.Cloud.define("joinQueue", function(request, response) {
                   }
                 }
               ).then( function(){
-                response.success(player);
+                sendDataChangedNotification().then(function() {
+                  response.success(player);
+                }).fail(function(err) {
+                  response.error(err);
+                });
```
Comment (udiWertheimer) [inline cloud/main.js:88]: @giladgo I'm not sure about this. Why would you not respond as "success" if the notification wasn't sent for some reason?

Comment (yonbergman) [inline cloud/main.js:88]: I agree with @udiWertheimer if sendNotification fails it shouldn't affect anything

File: cloud/main.js [hunk 2]
```diff
@@ -363,3 +377,14 @@ Parse.Cloud.define("endMatch", function(request, response) {
     }
   )
 });
+
+
+function sendDataChangedNotification() {
```
Comment (yonbergman) [inline cloud/main.js]: Can you Put this in a class please? for the cleanliness of the code..

**Comment by yonbergman** [comment]
:tiger: :moneybag:

File: cloud/main.js (hunk 1 above)
Comment (giladgo) [inline cloud/main.js:88]: :+1:

File: cloud/main.js [hunk 3]
```diff
@@ -1,4 +1,6 @@
 _ = require('underscore');
+notifications = require('cloud/notifications.js');
```
Comment (yonbergman) [inline cloud/main.js]: Notifications

File: cloud/notifications.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+  sendDataChangedNotification: function() {
```
Comment (yonbergman) [inline cloud/notifications.js]: `notifySessionDataChanged`

File: cloud/notifications.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    var everyoneQuery = new Parse.Query(Parse.Installation);
+    return Parse.Push.send({
+      where: everyoneQuery,
+      data: {
+        type: "dataChanged"
+      }
+    });
+  }
```
Comment (yonbergman) [inline cloud/notifications.js:15]: maybe catch failures here and log?

File: cloud/notifications.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+  sendDataChangedNotification: function() {
+    var everyoneQuery = new Parse.Query(Parse.Installation);
+    return Parse.Push.send({
+      where: everyoneQuery,
+      data: {
+        type: "dataChanged"
```
Comment (yonbergman) [inline cloud/notifications.js]: sessionDataChanged and constantize

File: cloud/main.js [hunk 7]
```diff
@@ -308,7 +316,11 @@ Parse.Cloud.define("setCharacter", function(request, response) {
       player.set("characterB", charB);
       return player.save();
     }
-  ).done(
+  ).then(function(player) {
+    return notifications.sendDataChangedNotification().done(function() {
```
Comment (yonbergman) [inline cloud/main.js]: change to then

File: cloud/main.js [hunk 8]
```diff
@@ -353,7 +365,11 @@ Parse.Cloud.define("endMatch", function(request, response) {
         Match.startMatch(winner).then(
           function(newMatch){
             if (newMatch){
-              response.success(newMatch);
+              notifications.sendDataChangedNotification().done(function() {
```
Comment (yonbergman) [inline cloud/main.js]: then and remove fail

File: cloud/main.js [hunk 9]
```diff
@@ -269,7 +275,9 @@ Parse.Cloud.define("leaveQueue", function(request, response) {
       } else {
         queueItem.dequeue().done(
           function(player){
-            response.success();
+            return notifications.sendDataChangedNotification().done(function() {
```
Comment (yonbergman) [inline cloud/main.js]: then

File: cloud/main.js [hunk 10]
```diff
@@ -237,7 +239,11 @@ Parse.Cloud.define("joinQueue", function(request, response) {
                   }
                 }
               ).then( function(){
-                response.success(player);
+                notifications.sendDataChangedNotification().then(function() {
+                  response.success(player);
+                }).fail(function(err) {
```
Comment (yonbergman) [inline cloud/main.js]: remove fail

File: cloud/main.js (hunk 3 above)
Comment (giladgo) [inline cloud/main.js]: :+1:

File: cloud/notifications.js (hunk 4 above)
Comment (giladgo) [inline cloud/notifications.js]: :+1:

File: cloud/notifications.js (hunk 5 above)
Comment (giladgo) [inline cloud/notifications.js:15]: :+1:

File: cloud/notifications.js (hunk 6 above)
Comment (giladgo) [inline cloud/notifications.js]: :+1:

File: cloud/main.js (hunk 7 above)
Comment (giladgo) [inline cloud/main.js]: :+1:

File: cloud/main.js (hunk 9 above)
Comment (giladgo) [inline cloud/main.js]: :+1:

File: cloud/main.js (hunk 10 above)
Comment (giladgo) [inline cloud/main.js]: :+1:

File: cloud/main.js [hunk 11]
```diff
@@ -114,7 +117,9 @@ Parse.Cloud.define("leaveQueue", function(request, response) {
       } else {
         queueItem.dequeue().done(
           function(player){
-            response.success();
+            return Notifications.notifySessionDataChanged().done(function() {
```
Comment (yonbergman) [inline cloud/main.js]: then

**Pull request merged**